src/Geno/C++/GUI/Widgets/TextEdit.cpp

```diff
@@ -218,9 +238,1592 @@ void TextEdit::OnDragDrop( const Drop& rDrop, int X, int Y )
+
+bool TextEdit::IsCoordinateInText( File& rFile, Coordinate Position )
+{
+	if( Position.x == 0 ) return false;
+
+	Coordinate Start;
+
+	GetWordAt( rFile, Coordinate( Position.x - 1, Position.y ), &Start, nullptr );
+
+	if( Start.x == 0 ) return false;
+
+	return true;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::AdjustCursorIfInText( File& rFile, Cursor& rCursor, int LineIndex, int XOffset )
+{
+	if( IsCoordinateInText( rFile, rCursor.Position ) && LineIndex == rCursor.Position.y ) rCursor.Position.x += XOffset;
+	if( IsCoordinateInText( rFile, rCursor.SelectionOrigin ) && LineIndex == rCursor.SelectionOrigin.y ) rCursor.SelectionOrigin.x += XOffset;
+	if( IsCoordinateInText( rFile, rCursor.SelectionStart ) && LineIndex == rCursor.SelectionStart.y ) rCursor.SelectionStart.x += XOffset;
+	if( IsCoordinateInText( rFile, rCursor.SelectionEnd ) && LineIndex == rCursor.SelectionEnd.y ) rCursor.SelectionEnd.x += XOffset;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::AdjustCursor( File& rFile, Cursor& rCursor, int XOffset )
+{
+	rCursor.Position.x += XOffset;
+	rCursor.SelectionOrigin.x += XOffset;
+	rCursor.SelectionStart.x += XOffset;
+	rCursor.SelectionEnd.x += XOffset;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::SetSelectionLine( File& rFile, int LineIndex )
+{
+	if( LineIndex >= rFile.Lines.size() ) return;
+
+	Line& rLine = rFile.Lines[ LineIndex ];
+
+	Coordinate Start( 0, LineIndex );
+	Coordinate End( ( int )rLine.size(), LineIndex );
+
+	SetSelection( rFile, Start, End, 0 );
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+void TextEdit::SetSelection( File& rFile, Coordinate Start, Coordinate End, int CursorIndex )
+{
+	GENO_ASSERT( Start.x < End.x && Start.y < End.y );
+	GENO_ASSERT( CursorIndex < rFile.Cursors.size() );
+
+	auto& rLines = rFile.Lines;
+
+	if( Start.y >= rLines.size() ) return;
+
+	if( End.y >= rLines.size() )
+	{
+		End.y = ( int )rLines.size() - 1;
+
+		End.x = ( int )rLines[ End.y ].size();
+	}
+
+	Cursor& rCursor = rFile.Cursors[ CursorIndex ];
+
+	rCursor.SelectionStart = Start;
+	rCursor.SelectionEnd   = End;
+}
+
+//////////////////////////////////////////////////////////////////////////
+
+TextEdit::Coordinate TextEdit::GetCoordinate( File& rFile, ImVec2 Position, bool RelativeToEditor )
+{
+	if( !RelativeToEditor )
+	{
+		ImVec2 Origin = ImGui::GetCursorScreenPos();
+
+		Position.x -= Origin.x;
+		Position.y -= Origin.y;
+	}
+
+	int LineIndex = ( int )( ( ( Position.y + Props.ScrollY ) / Props.CharAdvanceY ) - ( Props.ScrollY / Props.CharAdvanceY ) );
+
+	int NumLines = ( int )rFile.Lines.size();
+
+	if( LineIndex > NumLines - 1 )
+	{
+		LineIndex = NumLines - 1;
+	}
+
+	const Line& rLine = rFile.Lines[ LineIndex ];
+
+	int LineSize = ( int )rLine.size();
+
+	char string[ 2 ] = { 0, 0 };
+
+	float length = 0.0f;
```

Comment:
You don't have to split variables up like this! I prefer bundling variables together and aligning them.
(Also check the naming convention in this particular example)

Just so that I don't have to put the same comment in several places, please overlook anywhere else you can bundle variables together (remember to format and align)